lib/remote_system.rb

```diff
@@ -16,16 +16,28 @@
 # you may find current contact information at www.suse.com
 
 class RemoteSystem < System
-  attr_accessor :host
-  attr_accessor :remote_user
+  attr_accessor :host, :remote_user, :ssh_port, :ssh_key_options
 
   def type
     "remote"
   end
 
-  def initialize(host, remote_user = "root")
+  def initialize(host, opts = {})
+    @options = {
+      remote_user: "root",
+      ssh_port: 22,
+      private_key_path: ""
+    }.merge(opts)
+
     @host = host
-    @remote_user = remote_user
+    @remote_user = @options[:remote_user]
+    @ssh_port = @options[:ssh_port]
+    @private_key_path = @options[:private_key_path]
+    @ssh_key_options = []
+
+    if @private_key_path.length > 0
+      @ssh_key_options = ["-i", @private_key_path]
```

Comment:
`@ssh_key_options` doesn't really make sense here as a name, as it's a generic array storing arbitrary options. I would rather store a list of identity files.